cmd/segment-reaper/observer_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/private/testcontext"
+	"storj.io/storj/private/testrand"
+)
+
+func Test_observer_analyzeProject(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	tests := []struct {
+		segments                 string
+		expectedNumberOfSegments byte
+		segmentsAfter            string
+	}{
+		// this visualize which segments will be NOT selected as zombie segments
+
+		// known number of segments
+		{"11111_l", 6, "11111_l"}, // #0
```

Comment:
Because of the change that I reported in the previous review (skip iterations when the mask is all set to 1), I've just thought that it would be good to add a test case for it.

Reply:
test case with `maxNumOfSegments`?

Reply:
yes, it's going to be a long line, but if you think that could be annoying and making the test harder to understand, then forget about it.

Reply:
I can pre-generate such string before test.

Reply:
It sounds good to me, I thought that but I didn't want to be too picky.